python/pylibcugraph/pylibcugraph/uniform_neighborhood_sampling.pyx

```diff
@@ -0,0 +1,190 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Have cython use python 3 syntax
+# cython: language_level = 3
+
+from libc.stdint cimport uintptr_t
+
+from pylibcugraph._cugraph_c.resource_handle cimport (
+    bool_t,
+    data_type_id_t,
+    cugraph_resource_handle_t,
+)
+from pylibcugraph._cugraph_c.error cimport (
+    cugraph_error_code_t,
+    cugraph_error_t,
+)
+from pylibcugraph._cugraph_c.array cimport (
+    cugraph_type_erased_device_array_view_t,
+    cugraph_type_erased_device_array_view_create,
+    cugraph_type_erased_device_array_free,
+    cugraph_type_erased_host_array_view_t,
+    cugraph_type_erased_host_array_view_create
+)
+from pylibcugraph._cugraph_c.graph cimport (
+    cugraph_graph_t,
+)
+from pylibcugraph._cugraph_c.algorithms cimport (
+    uniform_nbr_sample,
+    cugraph_sample_result_t,
+    cugraph_sample_result_get_sources,
+    cugraph_sample_result_get_destinations,
+    cugraph_sample_result_get_start_labels,
+    cugraph_sample_result_get_index,
+    cugraph_sample_result_get_counts,
+    cugraph_sample_result_free,
+)
+from pylibcugraph.resource_handle cimport (
+    EXPERIMENTAL__ResourceHandle,
+)
+from pylibcugraph.graphs cimport (
+    _GPUGraph
+)
+from pylibcugraph.utils cimport (
+    assert_success,
+    copy_to_cupy_array,
+    assert_CAI_type,
+    get_c_type_from_numpy_type,
+)
+
+def EXPERIMENTAL__uniform_neighborhood_sampling(EXPERIMENTAL__ResourceHandle resource_handle,
+                               _GPUGraph input_graph,
+                               start_info_list,
+                               h_fan_out,
+                               bool_t with_replacement,
+                               bool_t do_expensive_check):
+    """
+    Does uniform neighborhood sampling.
+
+    Parameters
+    ----------
+    resource_handle: ResourceHandle
+        Handle to the underlying device and host resources needed for
+        referencing data and running algorithms.
+
+    input_graph: MGGraph
+        The input graph, for Multi-GPU operations.
+
+    start_info_list: device array type
```

Comment:
My understanding is that there are 2 distinct variables (`start`, `start_label`) based on the C and C++ API. where is `start_info_list` coming from ?

Reply:
Maybe you are consolidating those two?

Reply:
Yes, in the issue tied to this PR, issue#2108, the API combines the start vertices along with their client labels into `start_info_list`, which is separated into `start` and `start_label` later on